<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>Test Report</title>
    <style>body {
  font-family: Helvetica, Arial, sans-serif;
  font-size: 12px;
  /* do not increase min-width as some may use split screens */
  min-width: 800px;
  color: #999;
}

h1 {
  font-size: 24px;
  color: black;
}

h2 {
  font-size: 16px;
  color: black;
}

p {
  color: black;
}

a {
  color: #999;
}

table {
  border-collapse: collapse;
}

/******************************
 * SUMMARY INFORMATION
 ******************************/
#environment td {
  padding: 5px;
  border: 1px solid #E6E6E6;
}
#environment tr:nth-child(odd) {
  background-color: #f6f6f6;
}

/******************************
 * TEST RESULT COLORS
 ******************************/
span.passed,
.passed .col-result {
  color: green;
}

span.skipped,
span.xfailed,
span.rerun,
.skipped .col-result,
.xfailed .col-result,
.rerun .col-result {
  color: orange;
}

span.error,
span.failed,
span.xpassed,
.error .col-result,
.failed .col-result,
.xpassed .col-result {
  color: red;
}

/******************************
 * RESULTS TABLE
 *
 * 1. Table Layout
 * 2. Extra
 * 3. Sorting items
 *
 ******************************/
/*------------------
 * 1. Table Layout
 *------------------*/
#results-table {
  border: 1px solid #e6e6e6;
  color: #999;
  font-size: 12px;
  width: 100%;
}
#results-table th,
#results-table td {
  padding: 5px;
  border: 1px solid #E6E6E6;
  text-align: left;
}
#results-table th {
  font-weight: bold;
}

/*------------------
 * 2. Extra
 *------------------*/
.log {
  background-color: #e6e6e6;
  border: 1px solid #e6e6e6;
  color: black;
  display: block;
  font-family: "Courier New", Courier, monospace;
  height: 230px;
  overflow-y: scroll;
  padding: 5px;
  white-space: pre-wrap;
}
.log:only-child {
  height: inherit;
}

div.image {
  border: 1px solid #e6e6e6;
  float: right;
  height: 240px;
  margin-left: 5px;
  overflow: hidden;
  width: 320px;
}
div.image img {
  width: 320px;
}

div.video {
  border: 1px solid #e6e6e6;
  float: right;
  height: 240px;
  margin-left: 5px;
  overflow: hidden;
  width: 320px;
}
div.video video {
  overflow: hidden;
  width: 320px;
  height: 240px;
}

.collapsed {
  display: none;
}

.expander::after {
  content: " (show details)";
  color: #BBB;
  font-style: italic;
  cursor: pointer;
}

.collapser::after {
  content: " (hide details)";
  color: #BBB;
  font-style: italic;
  cursor: pointer;
}

/*------------------
 * 3. Sorting items
 *------------------*/
.sortable {
  cursor: pointer;
}

.sort-icon {
  font-size: 0px;
  float: left;
  margin-right: 5px;
  margin-top: 5px;
  /*triangle*/
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
}
.inactive .sort-icon {
  /*finish triangle*/
  border-top: 8px solid #E6E6E6;
}
.asc.active .sort-icon {
  /*finish triangle*/
  border-bottom: 8px solid #999;
}
.desc.active .sort-icon {
  /*finish triangle*/
  border-top: 8px solid #999;
}
</style></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) { // eslint-disable-line no-redeclare
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function findAll(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sortColumn(elem) {
    toggleSortStates(elem);
    const colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    let key;
    if (elem.classList.contains('result')) {
        key = keyResult;
    } else if (elem.classList.contains('links')) {
        key = keyLink;
    } else {
        key = keyAlpha;
    }
    sortTable(elem, key(colIndex));
}

function showAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(showExtras);
}

function hideAllExtras() { // eslint-disable-line no-unused-vars
    findAll('.col-result').forEach(hideExtras);
}

function showExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.remove('collapsed');
    expandcollapse.classList.remove('expander');
    expandcollapse.classList.add('collapser');
}

function hideExtras(colresultElem) {
    const extras = colresultElem.parentNode.nextElementSibling;
    const expandcollapse = colresultElem.firstElementChild;
    extras.classList.add('collapsed');
    expandcollapse.classList.remove('collapser');
    expandcollapse.classList.add('expander');
}

function showFilters() {
    const filterItems = document.getElementsByClassName('filter');
    for (let i = 0; i < filterItems.length; i++)
        filterItems[i].hidden = false;
}

function addCollapse() {
    // Add links for show/hide all
    const resulttable = find('table#results-table');
    const showhideall = document.createElement('p');
    showhideall.innerHTML = '<a href="javascript:showAllExtras()">Show all details</a> / ' +
                            '<a href="javascript:hideAllExtras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    findAll('.col-result').forEach(function(elem) {
        const collapsed = getQueryParameter('collapsed') || 'Passed';
        const extras = elem.parentNode.nextElementSibling;
        const expandcollapse = document.createElement('span');
        if (extras.classList.contains('collapsed')) {
            expandcollapse.classList.add('expander');
        } else if (collapsed.includes(elem.innerHTML)) {
            extras.classList.add('collapsed');
            expandcollapse.classList.add('expander');
        } else {
            expandcollapse.classList.add('collapser');
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener('click', function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains('collapsed')) {
                showExtras(event.currentTarget);
            } else {
                hideExtras(event.currentTarget);
            }
        });
    });
}

function getQueryParameter(name) {
    const match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
}

function init () { // eslint-disable-line no-unused-vars
    resetSortHeaders();

    addCollapse();

    showFilters();

    sortColumn(find('.initial-sort'));

    findAll('.sortable').forEach(function(elem) {
        elem.addEventListener('click',
            function() {
                sortColumn(elem);
            }, false);
    });
}

function sortTable(clicked, keyFunc) {
    const rows = findAll('.results-table-row');
    const reversed = !clicked.classList.contains('asc');
    const sortedRows = sort(rows, keyFunc, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    const thead = document.getElementById('results-table-head');
    document.getElementById('results-table').remove();
    const parent = document.createElement('table');
    parent.id = 'results-table';
    parent.appendChild(thead);
    sortedRows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName('BODY')[0].appendChild(parent);
}

function sort(items, keyFunc, reversed) {
    const sortArray = items.map(function(item, i) {
        return [keyFunc(item), i];
    });

    sortArray.sort(function(a, b) {
        const keyA = a[0];
        const keyB = b[0];

        if (keyA == keyB) return 0;

        if (reversed) {
            return keyA < keyB ? 1 : -1;
        } else {
            return keyA > keyB ? 1 : -1;
        }
    });

    return sortArray.map(function(item) {
        const index = item[1];
        return items[index];
    });
}

function keyAlpha(colIndex) {
    return function(elem) {
        return elem.childNodes[1].childNodes[colIndex].firstChild.data.toLowerCase();
    };
}

function keyLink(colIndex) {
    return function(elem) {
        const dataCell = elem.childNodes[1].childNodes[colIndex].firstChild;
        return dataCell == null ? '' : dataCell.innerText.toLowerCase();
    };
}

function keyResult(colIndex) {
    return function(elem) {
        const strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
            'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[colIndex].firstChild.data);
    };
}

function resetSortHeaders() {
    findAll('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    findAll('.sortable').forEach(function(elem) {
        const icon = document.createElement('div');
        icon.className = 'sort-icon';
        icon.textContent = 'vvv';
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove('desc', 'active');
        elem.classList.add('asc', 'inactive');
    });
}

function toggleSortStates(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        resetSortHeaders();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function isAllRowsHidden(value) {
    return value.hidden == false;
}

function filterTable(elem) { // eslint-disable-line no-unused-vars
    const outcomeAtt = 'data-test-result';
    const outcome = elem.getAttribute(outcomeAtt);
    const classOutcome = outcome + ' results-table-row';
    const outcomeRows = document.getElementsByClassName(classOutcome);

    for(let i = 0; i < outcomeRows.length; i++){
        outcomeRows[i].hidden = !elem.checked;
    }

    const rows = findAll('.results-table-row').filter(isAllRowsHidden);
    const allRowsHidden = rows.length == 0 ? true : false;
    const notFoundMessage = document.getElementById('not-found-message');
    notFoundMessage.hidden = !allRowsHidden;
}
</script>
    <h1>自动化测试报告</h1>
    <p>Report generated on 24-Jul-2022 at 16:59:26 by <a href="https://pypi.python.org/pypi/pytest-html">pytest-html</a> v3.1.1</p>
    <h2>Environment</h2>
    <table id="environment">
      <tr>
        <td>Project Name</td>
        <td>test_project</td></tr>
      <tr>
        <td>Python</td>
        <td>3.8.10</td></tr></table>
    <h2>Summary</h2>
    <p>Tester: 123</p>
    <p>9 tests ran in 96.95 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="passed">7 passed</span>, <input checked="true" class="filter" data-test-result="skipped" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="skipped">1 skipped</span>, <input checked="true" class="filter" data-test-result="failed" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="failed">2 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filterTable(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th>Description</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable time" col="time">Execute Time</th>
          <th class="sortable" col="duration">Duration</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="5">No results found. Try to check the filters</th></tr></thead>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td> 测试002 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_002</td>
          <td class="col-time">2022-07-24 08:59:26.245161</td>
          <td class="col-duration">0.00</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[XPASS(strict)][gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_002 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.testcase.test_project.test_api:test_002:39 - 这是一个框架xfail修饰测试
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td> 测试004 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_004[module: test, case_id: event_query_002, case_desc: 测试全局变量, is_run: None, method: get, env: ${test}, url: /comments, params: {&apos;postId&apos;: 1}, headers: None,data_type: None,data: None,files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">22.80</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>conn = &lt;urllib3.connection.HTTPSConnection object at 0x000002D52AA0A220&gt;<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;<br/>timeout = Timeout(connect=10, read=10, total=None), chunked = False<br/>httplib_request_kw = {&#x27;body&#x27;: None, &#x27;headers&#x27;: {&#x27;User-Agent&#x27;: &#x27;python-requests/2.25.1&#x27;, &#x27;Accept-Encoding&#x27;: &#x27;gzip, deflate&#x27;, &#x27;Accept&#x27;: &#x27;*/*&#x27;, &#x27;Connection&#x27;: &#x27;keep-alive&#x27;}}<br/>timeout_obj = Timeout(connect=10, read=10, total=None), read_timeout = 10<br/><br/>    def _make_request(<br/>        self, conn, method, url, timeout=_Default, chunked=False, **httplib_request_kw<br/>    ):<br/>        &quot;&quot;&quot;<br/>        Perform a request on a given urllib connection object taken from our<br/>        pool.<br/>    <br/>        :param conn:<br/>            a connection from one of our connection pools<br/>    <br/>        :param timeout:<br/>            Socket timeout in seconds for the request. This can be a<br/>            float or integer, which will set the same timeout value for<br/>            the socket connect and the socket read, or an instance of<br/>            :class:`urllib3.util.Timeout`, which gives you more fine-grained<br/>            control over your timeouts.<br/>        &quot;&quot;&quot;<br/>        self.num_requests += 1<br/>    <br/>        timeout_obj = self._get_timeout(timeout)<br/>        timeout_obj.start_connect()<br/>        conn.timeout = timeout_obj.connect_timeout<br/>    <br/>        # Trigger any extra validation we need to do.<br/>        try:<br/>            self._validate_conn(conn)<br/>        except (SocketTimeout, BaseSSLError) as e:<br/>            # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.<br/>            self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)<br/>            raise<br/>    <br/>        # conn.request() calls http.client.*.request, not the method in<br/>        # urllib3.request. It also calls makefile (recv) on the socket.<br/>        try:<br/>            if chunked:<br/>                conn.request_chunked(method, url, **httplib_request_kw)<br/>            else:<br/>                conn.request(method, url, **httplib_request_kw)<br/>    <br/>        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is<br/>        # legitimately able to close the connection after sending a valid response.<br/>        # With this behaviour, the received response is still readable.<br/>        except BrokenPipeError:<br/>            # Python 3<br/>            pass<br/>        except IOError as e:<br/>            # Python 2 and macOS/Linux<br/>            # EPIPE and ESHUTDOWN are BrokenPipeError on Python 2, and EPROTOTYPE is needed on macOS<br/>            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/<br/>            if e.errno not in {<br/>                errno.EPIPE,<br/>                errno.ESHUTDOWN,<br/>                errno.EPROTOTYPE,<br/>            }:<br/>                raise<br/>    <br/>        # Reset the timeout for the recv() on the socket<br/>        read_timeout = timeout_obj.read_timeout<br/>    <br/>        # App Engine doesn&#x27;t have a sock attr<br/>        if getattr(conn, &quot;sock&quot;, None):<br/>            # In Python 3 socket.py will catch EAGAIN and return None when you<br/>            # try and read into the file pointer created by http.client, which<br/>            # instead raises a BadStatusLine exception. Instead of catching<br/>            # the exception and assuming all BadStatusLine exceptions are read<br/>            # timeouts, check for a zero timeout before making the request.<br/>            if read_timeout == 0:<br/>                raise ReadTimeoutError(<br/>                    self, url, &quot;Read timed out. (read timeout=%s)&quot; % read_timeout<br/>                )<br/>            if read_timeout is Timeout.DEFAULT_TIMEOUT:<br/>                conn.sock.settimeout(socket.getdefaulttimeout())<br/>            else:  # None or a value<br/>                conn.sock.settimeout(read_timeout)<br/>    <br/>        # Receive the response from the server<br/>        try:<br/>            try:<br/>                # Python 2.7, use buffering of HTTP responses<br/>                httplib_response = conn.getresponse(buffering=True)<br/>            except TypeError:<br/>                # Python 3<br/>                try:<br/>                    httplib_response = conn.getresponse()<br/>                except BaseException as e:<br/>                    # Remove the TypeError from the exception chain in<br/>                    # Python 3 (including for exceptions like SystemExit).<br/>                    # Otherwise it looks like a bug in the code.<br/>&gt;                   six.raise_from(e, None)<br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:449: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>value = None, from_value = None<br/><br/>&gt;   ???<br/><br/>&lt;string&gt;:3: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>conn = &lt;urllib3.connection.HTTPSConnection object at 0x000002D52AA0A220&gt;<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;<br/>timeout = Timeout(connect=10, read=10, total=None), chunked = False<br/>httplib_request_kw = {&#x27;body&#x27;: None, &#x27;headers&#x27;: {&#x27;User-Agent&#x27;: &#x27;python-requests/2.25.1&#x27;, &#x27;Accept-Encoding&#x27;: &#x27;gzip, deflate&#x27;, &#x27;Accept&#x27;: &#x27;*/*&#x27;, &#x27;Connection&#x27;: &#x27;keep-alive&#x27;}}<br/>timeout_obj = Timeout(connect=10, read=10, total=None), read_timeout = 10<br/><br/>    def _make_request(<br/>        self, conn, method, url, timeout=_Default, chunked=False, **httplib_request_kw<br/>    ):<br/>        &quot;&quot;&quot;<br/>        Perform a request on a given urllib connection object taken from our<br/>        pool.<br/>    <br/>        :param conn:<br/>            a connection from one of our connection pools<br/>    <br/>        :param timeout:<br/>            Socket timeout in seconds for the request. This can be a<br/>            float or integer, which will set the same timeout value for<br/>            the socket connect and the socket read, or an instance of<br/>            :class:`urllib3.util.Timeout`, which gives you more fine-grained<br/>            control over your timeouts.<br/>        &quot;&quot;&quot;<br/>        self.num_requests += 1<br/>    <br/>        timeout_obj = self._get_timeout(timeout)<br/>        timeout_obj.start_connect()<br/>        conn.timeout = timeout_obj.connect_timeout<br/>    <br/>        # Trigger any extra validation we need to do.<br/>        try:<br/>            self._validate_conn(conn)<br/>        except (SocketTimeout, BaseSSLError) as e:<br/>            # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.<br/>            self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)<br/>            raise<br/>    <br/>        # conn.request() calls http.client.*.request, not the method in<br/>        # urllib3.request. It also calls makefile (recv) on the socket.<br/>        try:<br/>            if chunked:<br/>                conn.request_chunked(method, url, **httplib_request_kw)<br/>            else:<br/>                conn.request(method, url, **httplib_request_kw)<br/>    <br/>        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is<br/>        # legitimately able to close the connection after sending a valid response.<br/>        # With this behaviour, the received response is still readable.<br/>        except BrokenPipeError:<br/>            # Python 3<br/>            pass<br/>        except IOError as e:<br/>            # Python 2 and macOS/Linux<br/>            # EPIPE and ESHUTDOWN are BrokenPipeError on Python 2, and EPROTOTYPE is needed on macOS<br/>            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/<br/>            if e.errno not in {<br/>                errno.EPIPE,<br/>                errno.ESHUTDOWN,<br/>                errno.EPROTOTYPE,<br/>            }:<br/>                raise<br/>    <br/>        # Reset the timeout for the recv() on the socket<br/>        read_timeout = timeout_obj.read_timeout<br/>    <br/>        # App Engine doesn&#x27;t have a sock attr<br/>        if getattr(conn, &quot;sock&quot;, None):<br/>            # In Python 3 socket.py will catch EAGAIN and return None when you<br/>            # try and read into the file pointer created by http.client, which<br/>            # instead raises a BadStatusLine exception. Instead of catching<br/>            # the exception and assuming all BadStatusLine exceptions are read<br/>            # timeouts, check for a zero timeout before making the request.<br/>            if read_timeout == 0:<br/>                raise ReadTimeoutError(<br/>                    self, url, &quot;Read timed out. (read timeout=%s)&quot; % read_timeout<br/>                )<br/>            if read_timeout is Timeout.DEFAULT_TIMEOUT:<br/>                conn.sock.settimeout(socket.getdefaulttimeout())<br/>            else:  # None or a value<br/>                conn.sock.settimeout(read_timeout)<br/>    <br/>        # Receive the response from the server<br/>        try:<br/>            try:<br/>                # Python 2.7, use buffering of HTTP responses<br/>                httplib_response = conn.getresponse(buffering=True)<br/>            except TypeError:<br/>                # Python 3<br/>                try:<br/>&gt;                   httplib_response = conn.getresponse()<br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:444: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connection.HTTPSConnection object at 0x000002D52AA0A220&gt;<br/><br/>    def getresponse(self):<br/>        &quot;&quot;&quot;Get the response from the server.<br/>    <br/>        If the HTTPConnection is in the correct state, returns an<br/>        instance of HTTPResponse or of whatever object is returned by<br/>        the response_class variable.<br/>    <br/>        If a request has not been sent or if a previous response has<br/>        not be handled, ResponseNotReady is raised.  If the HTTP<br/>        response indicates that the connection should be closed, then<br/>        it will be closed before the response is returned.  When the<br/>        connection is closed, the underlying socket is closed.<br/>        &quot;&quot;&quot;<br/>    <br/>        # if a prior response has been completed, then forget about it.<br/>        if self.__response and self.__response.isclosed():<br/>            self.__response = None<br/>    <br/>        # if a prior response exists, then it must be completed (otherwise, we<br/>        # cannot read this response&#x27;s header to determine the connection-close<br/>        # behavior)<br/>        #<br/>        # note: if a prior response existed, but was connection-close, then the<br/>        # socket and response were made independent of this HTTPConnection<br/>        # object since a new request requires that we open a whole new<br/>        # connection<br/>        #<br/>        # this means the prior response had one of two states:<br/>        #   1) will_close: this connection was reset and the prior socket and<br/>        #                  response operate independently<br/>        #   2) persistent: the response was retained and we await its<br/>        #                  isclosed() status to become true.<br/>        #<br/>        if self.__state != _CS_REQ_SENT or self.__response:<br/>            raise ResponseNotReady(self.__state)<br/>    <br/>        if self.debuglevel &gt; 0:<br/>            response = self.response_class(self.sock, self.debuglevel,<br/>                                           method=self._method)<br/>        else:<br/>            response = self.response_class(self.sock, method=self._method)<br/>    <br/>        try:<br/>            try:<br/>&gt;               response.begin()<br/><br/>D:\Python3.8.10\lib\http\client.py:1344: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;http.client.HTTPResponse object at 0x000002D52AA0A6D0&gt;<br/><br/>    def begin(self):<br/>        if self.headers is not None:<br/>            # we&#x27;ve already started reading the response<br/>            return<br/>    <br/>        # read until we get a non-100 response<br/>        while True:<br/>&gt;           version, status, reason = self._read_status()<br/><br/>D:\Python3.8.10\lib\http\client.py:307: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;http.client.HTTPResponse object at 0x000002D52AA0A6D0&gt;<br/><br/>    def _read_status(self):<br/>&gt;       line = str(self.fp.readline(_MAXLINE + 1), &quot;iso-8859-1&quot;)<br/><br/>D:\Python3.8.10\lib\http\client.py:268: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;socket.SocketIO object at 0x000002D52AA0AC40&gt;<br/>b = &lt;memory at 0x000002D52AA20340&gt;<br/><br/>    def readinto(self, b):<br/>        &quot;&quot;&quot;Read up to len(b) bytes into the writable buffer *b* and return<br/>        the number of bytes read.  If the socket is non-blocking and no bytes<br/>        are available, None is returned.<br/>    <br/>        If *b* is non-empty, a 0 return value indicates that the connection<br/>        was shutdown at the other end.<br/>        &quot;&quot;&quot;<br/>        self._checkClosed()<br/>        self._checkReadable()<br/>        if self._timeout_occurred:<br/>            raise OSError(&quot;cannot read from timed out object&quot;)<br/>        while True:<br/>            try:<br/>&gt;               return self._sock.recv_into(b)<br/><br/>D:\Python3.8.10\lib\socket.py:669: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;ssl.SSLSocket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0&gt;<br/>buffer = &lt;memory at 0x000002D52AA20340&gt;, nbytes = 8192, flags = 0<br/><br/>    def recv_into(self, buffer, nbytes=None, flags=0):<br/>        self._checkClosed()<br/>        if buffer and (nbytes is None):<br/>            nbytes = len(buffer)<br/>        elif nbytes is None:<br/>            nbytes = 1024<br/>        if self._sslobj is not None:<br/>            if flags != 0:<br/>                raise ValueError(<br/>                  &quot;non-zero flags not allowed in calls to recv_into() on %s&quot; %<br/>                  self.__class__)<br/>&gt;           return self.read(nbytes, buffer)<br/><br/>D:\Python3.8.10\lib\ssl.py:1241: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;ssl.SSLSocket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0&gt;<br/>len = 8192, buffer = &lt;memory at 0x000002D52AA20340&gt;<br/><br/>    def read(self, len=1024, buffer=None):<br/>        &quot;&quot;&quot;Read up to LEN bytes and return them.<br/>        Return zero-length string on EOF.&quot;&quot;&quot;<br/>    <br/>        self._checkClosed()<br/>        if self._sslobj is None:<br/>            raise ValueError(&quot;Read on closed or unwrapped SSL socket.&quot;)<br/>        try:<br/>            if buffer is not None:<br/>&gt;               return self._sslobj.read(len, buffer)<br/><span class="error">E               socket.timeout: The read operation timed out</span><br/><br/>D:\Python3.8.10\lib\ssl.py:1099: timeout<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;requests.adapters.HTTPAdapter object at 0x000002D52AA0AE50&gt;<br/>request = &lt;PreparedRequest [GET]&gt;, stream = False<br/>timeout = Timeout(connect=10, read=10, total=None), verify = False, cert = None<br/>proxies = OrderedDict([(&#x27;ftp&#x27;, &#x27;ftp://127.0.0.1:10080&#x27;)])<br/><br/>    def send(self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None):<br/>        &quot;&quot;&quot;Sends PreparedRequest object. Returns Response object.<br/>    <br/>        :param request: The :class:`PreparedRequest &lt;PreparedRequest&gt;` being sent.<br/>        :param stream: (optional) Whether to stream the request content.<br/>        :param timeout: (optional) How long to wait for the server to send<br/>            data before giving up, as a float, or a :ref:`(connect timeout,<br/>            read timeout) &lt;timeouts&gt;` tuple.<br/>        :type timeout: float or tuple or urllib3 Timeout object<br/>        :param verify: (optional) Either a boolean, in which case it controls whether<br/>            we verify the server&#x27;s TLS certificate, or a string, in which case it<br/>            must be a path to a CA bundle to use<br/>        :param cert: (optional) Any user-provided SSL certificate to be trusted.<br/>        :param proxies: (optional) The proxies dictionary to apply to the request.<br/>        :rtype: requests.Response<br/>        &quot;&quot;&quot;<br/>    <br/>        try:<br/>            conn = self.get_connection(request.url, proxies)<br/>        except LocationValueError as e:<br/>            raise InvalidURL(e, request=request)<br/>    <br/>        self.cert_verify(conn, request.url, verify, cert)<br/>        url = self.request_url(request, proxies)<br/>        self.add_headers(request, stream=stream, timeout=timeout, verify=verify, cert=cert, proxies=proxies)<br/>    <br/>        chunked = not (request.body is None or &#x27;Content-Length&#x27; in request.headers)<br/>    <br/>        if isinstance(timeout, tuple):<br/>            try:<br/>                connect, read = timeout<br/>                timeout = TimeoutSauce(connect=connect, read=read)<br/>            except ValueError as e:<br/>                # this may raise a string formatting error.<br/>                err = (&quot;Invalid timeout {}. Pass a (connect, read) &quot;<br/>                       &quot;timeout tuple, or a single float to set &quot;<br/>                       &quot;both timeouts to the same value&quot;.format(timeout))<br/>                raise ValueError(err)<br/>        elif isinstance(timeout, TimeoutSauce):<br/>            pass<br/>        else:<br/>            timeout = TimeoutSauce(connect=timeout, read=timeout)<br/>    <br/>        try:<br/>            if not chunked:<br/>&gt;               resp = conn.urlopen(<br/>                    method=request.method,<br/>                    url=url,<br/>                    body=request.body,<br/>                    headers=request.headers,<br/>                    redirect=False,<br/>                    assert_same_host=False,<br/>                    preload_content=False,<br/>                    decode_content=False,<br/>                    retries=self.max_retries,<br/>                    timeout=timeout<br/>                )<br/><br/>..\venv\lib\site-packages\requests\adapters.py:439: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;, body = None<br/>headers = {&#x27;User-Agent&#x27;: &#x27;python-requests/2.25.1&#x27;, &#x27;Accept-Encoding&#x27;: &#x27;gzip, deflate&#x27;, &#x27;Accept&#x27;: &#x27;*/*&#x27;, &#x27;Connection&#x27;: &#x27;keep-alive&#x27;}<br/>retries = Retry(total=0, connect=None, read=False, redirect=None, status=None)<br/>redirect = False, assert_same_host = False<br/>timeout = Timeout(connect=10, read=10, total=None), pool_timeout = None<br/>release_conn = False, chunked = False, body_pos = None<br/>response_kw = {&#x27;decode_content&#x27;: False, &#x27;preload_content&#x27;: False}<br/>parsed_url = Url(scheme=None, auth=None, host=None, port=None, path=&#x27;/comments&#x27;, query=&#x27;postId=1&#x27;, fragment=None)<br/>destination_scheme = None, conn = None, release_this_conn = True<br/>http_tunnel_required = False, err = None, clean_exit = False<br/><br/>    def urlopen(<br/>        self,<br/>        method,<br/>        url,<br/>        body=None,<br/>        headers=None,<br/>        retries=None,<br/>        redirect=True,<br/>        assert_same_host=True,<br/>        timeout=_Default,<br/>        pool_timeout=None,<br/>        release_conn=None,<br/>        chunked=False,<br/>        body_pos=None,<br/>        **response_kw<br/>    ):<br/>        &quot;&quot;&quot;<br/>        Get a connection from the pool and perform an HTTP request. This is the<br/>        lowest level call for making a request, so you&#x27;ll need to specify all<br/>        the raw details.<br/>    <br/>        .. note::<br/>    <br/>           More commonly, it&#x27;s appropriate to use a convenience method provided<br/>           by :class:`.RequestMethods`, such as :meth:`request`.<br/>    <br/>        .. note::<br/>    <br/>           `release_conn` will only behave as expected if<br/>           `preload_content=False` because we want to make<br/>           `preload_content=False` the default behaviour someday soon without<br/>           breaking backwards compatibility.<br/>    <br/>        :param method:<br/>            HTTP request method (such as GET, POST, PUT, etc.)<br/>    <br/>        :param url:<br/>            The URL to perform the request on.<br/>    <br/>        :param body:<br/>            Data to send in the request body, either :class:`str`, :class:`bytes`,<br/>            an iterable of :class:`str`/:class:`bytes`, or a file-like object.<br/>    <br/>        :param headers:<br/>            Dictionary of custom headers to send, such as User-Agent,<br/>            If-None-Match, etc. If None, pool headers are used. If provided,<br/>            these headers completely replace any pool-specific headers.<br/>    <br/>        :param retries:<br/>            Configure the number of retries to allow before raising a<br/>            :class:`~urllib3.exceptions.MaxRetryError` exception.<br/>    <br/>            Pass ``None`` to retry until you receive a response. Pass a<br/>            :class:`~urllib3.util.retry.Retry` object for fine-grained control<br/>            over different types of retries.<br/>            Pass an integer number to retry connection errors that many times,<br/>            but no other types of errors. Pass zero to never retry.<br/>    <br/>            If ``False``, then retries are disabled and any exception is raised<br/>            immediately. Also, instead of raising a MaxRetryError on redirects,<br/>            the redirect response will be returned.<br/>    <br/>        :type retries: :class:`~urllib3.util.retry.Retry`, False, or an int.<br/>    <br/>        :param redirect:<br/>            If True, automatically handle redirects (status codes 301, 302,<br/>            303, 307, 308). Each redirect counts as a retry. Disabling retries<br/>            will disable redirect, too.<br/>    <br/>        :param assert_same_host:<br/>            If ``True``, will make sure that the host of the pool requests is<br/>            consistent else will raise HostChangedError. When ``False``, you can<br/>            use the pool on an HTTP proxy and request foreign hosts.<br/>    <br/>        :param timeout:<br/>            If specified, overrides the default timeout for this one<br/>            request. It may be a float (in seconds) or an instance of<br/>            :class:`urllib3.util.Timeout`.<br/>    <br/>        :param pool_timeout:<br/>            If set and the pool is set to block=True, then this method will<br/>            block for ``pool_timeout`` seconds and raise EmptyPoolError if no<br/>            connection is available within the time period.<br/>    <br/>        :param release_conn:<br/>            If False, then the urlopen call will not release the connection<br/>            back into the pool once a response is received (but will release if<br/>            you read the entire contents of the response such as when<br/>            `preload_content=True`). This is useful if you&#x27;re not preloading<br/>            the response&#x27;s content immediately. You will need to call<br/>            ``r.release_conn()`` on the response ``r`` to return the connection<br/>            back into the pool. If None, it takes the value of<br/>            ``response_kw.get(&#x27;preload_content&#x27;, True)``.<br/>    <br/>        :param chunked:<br/>            If True, urllib3 will send the body using chunked transfer<br/>            encoding. Otherwise, urllib3 will send the body using the standard<br/>            content-length form. Defaults to False.<br/>    <br/>        :param int body_pos:<br/>            Position to seek to in file-like body in the event of a retry or<br/>            redirect. Typically this won&#x27;t need to be set because urllib3 will<br/>            auto-populate the value when needed.<br/>    <br/>        :param \\**response_kw:<br/>            Additional parameters are passed to<br/>            :meth:`urllib3.response.HTTPResponse.from_httplib`<br/>        &quot;&quot;&quot;<br/>    <br/>        parsed_url = parse_url(url)<br/>        destination_scheme = parsed_url.scheme<br/>    <br/>        if headers is None:<br/>            headers = self.headers<br/>    <br/>        if not isinstance(retries, Retry):<br/>            retries = Retry.from_int(retries, redirect=redirect, default=self.retries)<br/>    <br/>        if release_conn is None:<br/>            release_conn = response_kw.get(&quot;preload_content&quot;, True)<br/>    <br/>        # Check host<br/>        if assert_same_host and not self.is_same_host(url):<br/>            raise HostChangedError(self, url, retries)<br/>    <br/>        # Ensure that the URL we&#x27;re connecting to is properly encoded<br/>        if url.startswith(&quot;/&quot;):<br/>            url = six.ensure_str(_encode_target(url))<br/>        else:<br/>            url = six.ensure_str(parsed_url.url)<br/>    <br/>        conn = None<br/>    <br/>        # Track whether `conn` needs to be released before<br/>        # returning/raising/recursing. Update this variable if necessary, and<br/>        # leave `release_conn` constant throughout the function. That way, if<br/>        # the function recurses, the original value of `release_conn` will be<br/>        # passed down into the recursive call, and its value will be respected.<br/>        #<br/>        # See issue #651 [1] for details.<br/>        #<br/>        # [1] &lt;https://github.com/urllib3/urllib3/issues/651&gt;<br/>        release_this_conn = release_conn<br/>    <br/>        http_tunnel_required = connection_requires_http_tunnel(<br/>            self.proxy, self.proxy_config, destination_scheme<br/>        )<br/>    <br/>        # Merge the proxy headers. Only done when not using HTTP CONNECT. We<br/>        # have to copy the headers dict so we can safely change it without those<br/>        # changes being reflected in anyone else&#x27;s copy.<br/>        if not http_tunnel_required:<br/>            headers = headers.copy()<br/>            headers.update(self.proxy_headers)<br/>    <br/>        # Must keep the exception bound to a separate variable or else Python 3<br/>        # complains about UnboundLocalError.<br/>        err = None<br/>    <br/>        # Keep track of whether we cleanly exited the except block. This<br/>        # ensures we do proper cleanup in finally.<br/>        clean_exit = False<br/>    <br/>        # Rewind body position, if needed. Record current position<br/>        # for future rewinds in the event of a redirect/retry.<br/>        body_pos = set_file_position(body, body_pos)<br/>    <br/>        try:<br/>            # Request a connection from the queue.<br/>            timeout_obj = self._get_timeout(timeout)<br/>            conn = self._get_conn(timeout=pool_timeout)<br/>    <br/>            conn.timeout = timeout_obj.connect_timeout<br/>    <br/>            is_new_proxy_conn = self.proxy is not None and not getattr(<br/>                conn, &quot;sock&quot;, None<br/>            )<br/>            if is_new_proxy_conn and http_tunnel_required:<br/>                self._prepare_proxy(conn)<br/>    <br/>            # Make the request on the httplib connection object.<br/>            httplib_response = self._make_request(<br/>                conn,<br/>                method,<br/>                url,<br/>                timeout=timeout_obj,<br/>                body=body,<br/>                headers=headers,<br/>                chunked=chunked,<br/>            )<br/>    <br/>            # If we&#x27;re going to release the connection in ``finally:``, then<br/>            # the response doesn&#x27;t need to know about the connection. Otherwise<br/>            # it will also try to release it and we&#x27;ll have a double-release<br/>            # mess.<br/>            response_conn = conn if not release_conn else None<br/>    <br/>            # Pass method to Response for length checking<br/>            response_kw[&quot;request_method&quot;] = method<br/>    <br/>            # Import httplib&#x27;s response into our own wrapper object<br/>            response = self.ResponseCls.from_httplib(<br/>                httplib_response,<br/>                pool=self,<br/>                connection=response_conn,<br/>                retries=retries,<br/>                **response_kw<br/>            )<br/>    <br/>            # Everything went great!<br/>            clean_exit = True<br/>    <br/>        except EmptyPoolError:<br/>            # Didn&#x27;t get a connection from the pool, no need to clean up<br/>            clean_exit = True<br/>            release_this_conn = False<br/>            raise<br/>    <br/>        except (<br/>            TimeoutError,<br/>            HTTPException,<br/>            SocketError,<br/>            ProtocolError,<br/>            BaseSSLError,<br/>            SSLError,<br/>            CertificateError,<br/>        ) as e:<br/>            # Discard the connection for these exceptions. It will be<br/>            # replaced during the next _get_conn() call.<br/>            clean_exit = False<br/>    <br/>            def _is_ssl_error_message_from_http_proxy(ssl_error):<br/>                # We&#x27;re trying to detect the message &#x27;WRONG_VERSION_NUMBER&#x27; but<br/>                # SSLErrors are kinda all over the place when it comes to the message,<br/>                # so we try to cover our bases here!<br/>                message = &quot; &quot;.join(re.split(&quot;[^a-z]&quot;, str(ssl_error).lower()))<br/>                return (<br/>                    &quot;wrong version number&quot; in message or &quot;unknown protocol&quot; in message<br/>                )<br/>    <br/>            # Try to detect a common user error with proxies which is to<br/>            # set an HTTP proxy to be HTTPS when it should be &#x27;http://&#x27;<br/>            # (ie {&#x27;http&#x27;: &#x27;http://proxy&#x27;, &#x27;https&#x27;: &#x27;https://proxy&#x27;})<br/>            # Instead we add a nice error message and point to a URL.<br/>            if (<br/>                isinstance(e, BaseSSLError)<br/>                and self.proxy<br/>                and _is_ssl_error_message_from_http_proxy(e)<br/>                and conn.proxy<br/>                and conn.proxy.scheme == &quot;https&quot;<br/>            ):<br/>                e = ProxyError(<br/>                    &quot;Your proxy appears to only use HTTP and not HTTPS, &quot;<br/>                    &quot;try changing your proxy URL to be HTTP. See: &quot;<br/>                    &quot;https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html&quot;<br/>                    &quot;#https-proxy-error-http-proxy&quot;,<br/>                    SSLError(e),<br/>                )<br/>            elif isinstance(e, (BaseSSLError, CertificateError)):<br/>                e = SSLError(e)<br/>            elif isinstance(e, (SocketError, NewConnectionError)) and self.proxy:<br/>                e = ProxyError(&quot;Cannot connect to proxy.&quot;, e)<br/>            elif isinstance(e, (SocketError, HTTPException)):<br/>                e = ProtocolError(&quot;Connection aborted.&quot;, e)<br/>    <br/>&gt;           retries = retries.increment(<br/>                method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]<br/>            )<br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:787: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = Retry(total=0, connect=None, read=False, redirect=None, status=None)<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;, response = None<br/>error = ReadTimeoutError(&quot;HTTPSConnectionPool(host=&#x27;jsonplaceholder.typicode.com&#x27;, port=443): Read timed out. (read timeout=10)&quot;)<br/>_pool = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>_stacktrace = &lt;traceback object at 0x000002D52A9F9BC0&gt;<br/><br/>    def increment(<br/>        self,<br/>        method=None,<br/>        url=None,<br/>        response=None,<br/>        error=None,<br/>        _pool=None,<br/>        _stacktrace=None,<br/>    ):<br/>        &quot;&quot;&quot;Return a new Retry object with incremented retry counters.<br/>    <br/>        :param response: A response object, or None, if the server did not<br/>            return a response.<br/>        :type response: :class:`~urllib3.response.HTTPResponse`<br/>        :param Exception error: An error encountered during the request, or<br/>            None if the response was received successfully.<br/>    <br/>        :return: A new ``Retry`` object.<br/>        &quot;&quot;&quot;<br/>        if self.total is False and error:<br/>            # Disabled, indicate to re-raise the error.<br/>            raise six.reraise(type(error), error, _stacktrace)<br/>    <br/>        total = self.total<br/>        if total is not None:<br/>            total -= 1<br/>    <br/>        connect = self.connect<br/>        read = self.read<br/>        redirect = self.redirect<br/>        status_count = self.status<br/>        other = self.other<br/>        cause = &quot;unknown&quot;<br/>        status = None<br/>        redirect_location = None<br/>    <br/>        if error and self._is_connection_error(error):<br/>            # Connect retry?<br/>            if connect is False:<br/>                raise six.reraise(type(error), error, _stacktrace)<br/>            elif connect is not None:<br/>                connect -= 1<br/>    <br/>        elif error and self._is_read_error(error):<br/>            # Read retry?<br/>            if read is False or not self._is_method_retryable(method):<br/>&gt;               raise six.reraise(type(error), error, _stacktrace)<br/><br/>..\venv\lib\site-packages\urllib3\util\retry.py:550: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>tp = &lt;class &#x27;urllib3.exceptions.ReadTimeoutError&#x27;&gt;, value = None, tb = None<br/><br/>    def reraise(tp, value, tb=None):<br/>        try:<br/>            if value is None:<br/>                value = tp()<br/>            if value.__traceback__ is not tb:<br/>                raise value.with_traceback(tb)<br/>&gt;           raise value<br/><br/>..\venv\lib\site-packages\urllib3\packages\six.py:770: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;, body = None<br/>headers = {&#x27;User-Agent&#x27;: &#x27;python-requests/2.25.1&#x27;, &#x27;Accept-Encoding&#x27;: &#x27;gzip, deflate&#x27;, &#x27;Accept&#x27;: &#x27;*/*&#x27;, &#x27;Connection&#x27;: &#x27;keep-alive&#x27;}<br/>retries = Retry(total=0, connect=None, read=False, redirect=None, status=None)<br/>redirect = False, assert_same_host = False<br/>timeout = Timeout(connect=10, read=10, total=None), pool_timeout = None<br/>release_conn = False, chunked = False, body_pos = None<br/>response_kw = {&#x27;decode_content&#x27;: False, &#x27;preload_content&#x27;: False}<br/>parsed_url = Url(scheme=None, auth=None, host=None, port=None, path=&#x27;/comments&#x27;, query=&#x27;postId=1&#x27;, fragment=None)<br/>destination_scheme = None, conn = None, release_this_conn = True<br/>http_tunnel_required = False, err = None, clean_exit = False<br/><br/>    def urlopen(<br/>        self,<br/>        method,<br/>        url,<br/>        body=None,<br/>        headers=None,<br/>        retries=None,<br/>        redirect=True,<br/>        assert_same_host=True,<br/>        timeout=_Default,<br/>        pool_timeout=None,<br/>        release_conn=None,<br/>        chunked=False,<br/>        body_pos=None,<br/>        **response_kw<br/>    ):<br/>        &quot;&quot;&quot;<br/>        Get a connection from the pool and perform an HTTP request. This is the<br/>        lowest level call for making a request, so you&#x27;ll need to specify all<br/>        the raw details.<br/>    <br/>        .. note::<br/>    <br/>           More commonly, it&#x27;s appropriate to use a convenience method provided<br/>           by :class:`.RequestMethods`, such as :meth:`request`.<br/>    <br/>        .. note::<br/>    <br/>           `release_conn` will only behave as expected if<br/>           `preload_content=False` because we want to make<br/>           `preload_content=False` the default behaviour someday soon without<br/>           breaking backwards compatibility.<br/>    <br/>        :param method:<br/>            HTTP request method (such as GET, POST, PUT, etc.)<br/>    <br/>        :param url:<br/>            The URL to perform the request on.<br/>    <br/>        :param body:<br/>            Data to send in the request body, either :class:`str`, :class:`bytes`,<br/>            an iterable of :class:`str`/:class:`bytes`, or a file-like object.<br/>    <br/>        :param headers:<br/>            Dictionary of custom headers to send, such as User-Agent,<br/>            If-None-Match, etc. If None, pool headers are used. If provided,<br/>            these headers completely replace any pool-specific headers.<br/>    <br/>        :param retries:<br/>            Configure the number of retries to allow before raising a<br/>            :class:`~urllib3.exceptions.MaxRetryError` exception.<br/>    <br/>            Pass ``None`` to retry until you receive a response. Pass a<br/>            :class:`~urllib3.util.retry.Retry` object for fine-grained control<br/>            over different types of retries.<br/>            Pass an integer number to retry connection errors that many times,<br/>            but no other types of errors. Pass zero to never retry.<br/>    <br/>            If ``False``, then retries are disabled and any exception is raised<br/>            immediately. Also, instead of raising a MaxRetryError on redirects,<br/>            the redirect response will be returned.<br/>    <br/>        :type retries: :class:`~urllib3.util.retry.Retry`, False, or an int.<br/>    <br/>        :param redirect:<br/>            If True, automatically handle redirects (status codes 301, 302,<br/>            303, 307, 308). Each redirect counts as a retry. Disabling retries<br/>            will disable redirect, too.<br/>    <br/>        :param assert_same_host:<br/>            If ``True``, will make sure that the host of the pool requests is<br/>            consistent else will raise HostChangedError. When ``False``, you can<br/>            use the pool on an HTTP proxy and request foreign hosts.<br/>    <br/>        :param timeout:<br/>            If specified, overrides the default timeout for this one<br/>            request. It may be a float (in seconds) or an instance of<br/>            :class:`urllib3.util.Timeout`.<br/>    <br/>        :param pool_timeout:<br/>            If set and the pool is set to block=True, then this method will<br/>            block for ``pool_timeout`` seconds and raise EmptyPoolError if no<br/>            connection is available within the time period.<br/>    <br/>        :param release_conn:<br/>            If False, then the urlopen call will not release the connection<br/>            back into the pool once a response is received (but will release if<br/>            you read the entire contents of the response such as when<br/>            `preload_content=True`). This is useful if you&#x27;re not preloading<br/>            the response&#x27;s content immediately. You will need to call<br/>            ``r.release_conn()`` on the response ``r`` to return the connection<br/>            back into the pool. If None, it takes the value of<br/>            ``response_kw.get(&#x27;preload_content&#x27;, True)``.<br/>    <br/>        :param chunked:<br/>            If True, urllib3 will send the body using chunked transfer<br/>            encoding. Otherwise, urllib3 will send the body using the standard<br/>            content-length form. Defaults to False.<br/>    <br/>        :param int body_pos:<br/>            Position to seek to in file-like body in the event of a retry or<br/>            redirect. Typically this won&#x27;t need to be set because urllib3 will<br/>            auto-populate the value when needed.<br/>    <br/>        :param \\**response_kw:<br/>            Additional parameters are passed to<br/>            :meth:`urllib3.response.HTTPResponse.from_httplib`<br/>        &quot;&quot;&quot;<br/>    <br/>        parsed_url = parse_url(url)<br/>        destination_scheme = parsed_url.scheme<br/>    <br/>        if headers is None:<br/>            headers = self.headers<br/>    <br/>        if not isinstance(retries, Retry):<br/>            retries = Retry.from_int(retries, redirect=redirect, default=self.retries)<br/>    <br/>        if release_conn is None:<br/>            release_conn = response_kw.get(&quot;preload_content&quot;, True)<br/>    <br/>        # Check host<br/>        if assert_same_host and not self.is_same_host(url):<br/>            raise HostChangedError(self, url, retries)<br/>    <br/>        # Ensure that the URL we&#x27;re connecting to is properly encoded<br/>        if url.startswith(&quot;/&quot;):<br/>            url = six.ensure_str(_encode_target(url))<br/>        else:<br/>            url = six.ensure_str(parsed_url.url)<br/>    <br/>        conn = None<br/>    <br/>        # Track whether `conn` needs to be released before<br/>        # returning/raising/recursing. Update this variable if necessary, and<br/>        # leave `release_conn` constant throughout the function. That way, if<br/>        # the function recurses, the original value of `release_conn` will be<br/>        # passed down into the recursive call, and its value will be respected.<br/>        #<br/>        # See issue #651 [1] for details.<br/>        #<br/>        # [1] &lt;https://github.com/urllib3/urllib3/issues/651&gt;<br/>        release_this_conn = release_conn<br/>    <br/>        http_tunnel_required = connection_requires_http_tunnel(<br/>            self.proxy, self.proxy_config, destination_scheme<br/>        )<br/>    <br/>        # Merge the proxy headers. Only done when not using HTTP CONNECT. We<br/>        # have to copy the headers dict so we can safely change it without those<br/>        # changes being reflected in anyone else&#x27;s copy.<br/>        if not http_tunnel_required:<br/>            headers = headers.copy()<br/>            headers.update(self.proxy_headers)<br/>    <br/>        # Must keep the exception bound to a separate variable or else Python 3<br/>        # complains about UnboundLocalError.<br/>        err = None<br/>    <br/>        # Keep track of whether we cleanly exited the except block. This<br/>        # ensures we do proper cleanup in finally.<br/>        clean_exit = False<br/>    <br/>        # Rewind body position, if needed. Record current position<br/>        # for future rewinds in the event of a redirect/retry.<br/>        body_pos = set_file_position(body, body_pos)<br/>    <br/>        try:<br/>            # Request a connection from the queue.<br/>            timeout_obj = self._get_timeout(timeout)<br/>            conn = self._get_conn(timeout=pool_timeout)<br/>    <br/>            conn.timeout = timeout_obj.connect_timeout<br/>    <br/>            is_new_proxy_conn = self.proxy is not None and not getattr(<br/>                conn, &quot;sock&quot;, None<br/>            )<br/>            if is_new_proxy_conn and http_tunnel_required:<br/>                self._prepare_proxy(conn)<br/>    <br/>            # Make the request on the httplib connection object.<br/>&gt;           httplib_response = self._make_request(<br/>                conn,<br/>                method,<br/>                url,<br/>                timeout=timeout_obj,<br/>                body=body,<br/>                headers=headers,<br/>                chunked=chunked,<br/>            )<br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:703: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>conn = &lt;urllib3.connection.HTTPSConnection object at 0x000002D52AA0A220&gt;<br/>method = &#x27;GET&#x27;, url = &#x27;/comments?postId=1&#x27;<br/>timeout = Timeout(connect=10, read=10, total=None), chunked = False<br/>httplib_request_kw = {&#x27;body&#x27;: None, &#x27;headers&#x27;: {&#x27;User-Agent&#x27;: &#x27;python-requests/2.25.1&#x27;, &#x27;Accept-Encoding&#x27;: &#x27;gzip, deflate&#x27;, &#x27;Accept&#x27;: &#x27;*/*&#x27;, &#x27;Connection&#x27;: &#x27;keep-alive&#x27;}}<br/>timeout_obj = Timeout(connect=10, read=10, total=None), read_timeout = 10<br/><br/>    def _make_request(<br/>        self, conn, method, url, timeout=_Default, chunked=False, **httplib_request_kw<br/>    ):<br/>        &quot;&quot;&quot;<br/>        Perform a request on a given urllib connection object taken from our<br/>        pool.<br/>    <br/>        :param conn:<br/>            a connection from one of our connection pools<br/>    <br/>        :param timeout:<br/>            Socket timeout in seconds for the request. This can be a<br/>            float or integer, which will set the same timeout value for<br/>            the socket connect and the socket read, or an instance of<br/>            :class:`urllib3.util.Timeout`, which gives you more fine-grained<br/>            control over your timeouts.<br/>        &quot;&quot;&quot;<br/>        self.num_requests += 1<br/>    <br/>        timeout_obj = self._get_timeout(timeout)<br/>        timeout_obj.start_connect()<br/>        conn.timeout = timeout_obj.connect_timeout<br/>    <br/>        # Trigger any extra validation we need to do.<br/>        try:<br/>            self._validate_conn(conn)<br/>        except (SocketTimeout, BaseSSLError) as e:<br/>            # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.<br/>            self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)<br/>            raise<br/>    <br/>        # conn.request() calls http.client.*.request, not the method in<br/>        # urllib3.request. It also calls makefile (recv) on the socket.<br/>        try:<br/>            if chunked:<br/>                conn.request_chunked(method, url, **httplib_request_kw)<br/>            else:<br/>                conn.request(method, url, **httplib_request_kw)<br/>    <br/>        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is<br/>        # legitimately able to close the connection after sending a valid response.<br/>        # With this behaviour, the received response is still readable.<br/>        except BrokenPipeError:<br/>            # Python 3<br/>            pass<br/>        except IOError as e:<br/>            # Python 2 and macOS/Linux<br/>            # EPIPE and ESHUTDOWN are BrokenPipeError on Python 2, and EPROTOTYPE is needed on macOS<br/>            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/<br/>            if e.errno not in {<br/>                errno.EPIPE,<br/>                errno.ESHUTDOWN,<br/>                errno.EPROTOTYPE,<br/>            }:<br/>                raise<br/>    <br/>        # Reset the timeout for the recv() on the socket<br/>        read_timeout = timeout_obj.read_timeout<br/>    <br/>        # App Engine doesn&#x27;t have a sock attr<br/>        if getattr(conn, &quot;sock&quot;, None):<br/>            # In Python 3 socket.py will catch EAGAIN and return None when you<br/>            # try and read into the file pointer created by http.client, which<br/>            # instead raises a BadStatusLine exception. Instead of catching<br/>            # the exception and assuming all BadStatusLine exceptions are read<br/>            # timeouts, check for a zero timeout before making the request.<br/>            if read_timeout == 0:<br/>                raise ReadTimeoutError(<br/>                    self, url, &quot;Read timed out. (read timeout=%s)&quot; % read_timeout<br/>                )<br/>            if read_timeout is Timeout.DEFAULT_TIMEOUT:<br/>                conn.sock.settimeout(socket.getdefaulttimeout())<br/>            else:  # None or a value<br/>                conn.sock.settimeout(read_timeout)<br/>    <br/>        # Receive the response from the server<br/>        try:<br/>            try:<br/>                # Python 2.7, use buffering of HTTP responses<br/>                httplib_response = conn.getresponse(buffering=True)<br/>            except TypeError:<br/>                # Python 3<br/>                try:<br/>                    httplib_response = conn.getresponse()<br/>                except BaseException as e:<br/>                    # Remove the TypeError from the exception chain in<br/>                    # Python 3 (including for exceptions like SystemExit).<br/>                    # Otherwise it looks like a bug in the code.<br/>                    six.raise_from(e, None)<br/>        except (SocketTimeout, BaseSSLError, SocketError) as e:<br/>&gt;           self._raise_timeout(err=e, url=url, timeout_value=read_timeout)<br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:451: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;urllib3.connectionpool.HTTPSConnectionPool object at 0x000002D52A9F44F0&gt;<br/>err = timeout(&#x27;The read operation timed out&#x27;), url = &#x27;/comments?postId=1&#x27;<br/>timeout_value = 10<br/><br/>    def _raise_timeout(self, err, url, timeout_value):<br/>        &quot;&quot;&quot;Is the error actually a timeout? Will raise a ReadTimeout or pass&quot;&quot;&quot;<br/>    <br/>        if isinstance(err, SocketTimeout):<br/>&gt;           raise ReadTimeoutError(<br/>                self, url, &quot;Read timed out. (read timeout=%s)&quot; % timeout_value<br/>            )<br/><span class="error">E           urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host=&#x27;jsonplaceholder.typicode.com&#x27;, port=443): Read timed out. (read timeout=10)</span><br/><br/>..\venv\lib\site-packages\urllib3\connectionpool.py:340: ReadTimeoutError<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;fastpt.testcase.test_project.test_api.TestDemo object at 0x000002D52A9FE940&gt;<br/>data = {&#x27;assert&#x27;: None, &#x27;case_desc&#x27;: &#x27;测试全局变量&#x27;, &#x27;case_id&#x27;: &#x27;event_query_002&#x27;, &#x27;data&#x27;: None, ...}<br/><br/>    @allure.story(&quot;yaml数据测试输出&quot;)<br/>    @pytest.mark.test_api<br/>    @pytest.mark.parametrize(&#x27;data&#x27;, yaml_data, ids=yaml_ids)<br/>    def test_004(self, data):<br/>        &quot;&quot;&quot; 测试004 &quot;&quot;&quot;<br/>&gt;       rp = send_request.send_request(data)<br/><br/>testcase\test_project\test_api.py:66: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/>common\send_request.py:123: in send_request<br/>    response = self.__requests_engin(**parsed_data.get_request_args_parsed, **kwargs)<br/>common\send_request.py:73: in __requests_engin<br/>    raise e<br/>common\send_request.py:64: in __requests_engin<br/>    response = requests.session().request(<br/>..\venv\lib\site-packages\requests\sessions.py:542: in request<br/>    resp = self.send(prep, **send_kwargs)<br/>..\venv\lib\site-packages\requests\sessions.py:655: in send<br/>    r = adapter.send(request, **kwargs)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _<br/><br/>self = &lt;requests.adapters.HTTPAdapter object at 0x000002D52AA0AE50&gt;<br/>request = &lt;PreparedRequest [GET]&gt;, stream = False<br/>timeout = Timeout(connect=10, read=10, total=None), verify = False, cert = None<br/>proxies = OrderedDict([(&#x27;ftp&#x27;, &#x27;ftp://127.0.0.1:10080&#x27;)])<br/><br/>    def send(self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None):<br/>        &quot;&quot;&quot;Sends PreparedRequest object. Returns Response object.<br/>    <br/>        :param request: The :class:`PreparedRequest &lt;PreparedRequest&gt;` being sent.<br/>        :param stream: (optional) Whether to stream the request content.<br/>        :param timeout: (optional) How long to wait for the server to send<br/>            data before giving up, as a float, or a :ref:`(connect timeout,<br/>            read timeout) &lt;timeouts&gt;` tuple.<br/>        :type timeout: float or tuple or urllib3 Timeout object<br/>        :param verify: (optional) Either a boolean, in which case it controls whether<br/>            we verify the server&#x27;s TLS certificate, or a string, in which case it<br/>            must be a path to a CA bundle to use<br/>        :param cert: (optional) Any user-provided SSL certificate to be trusted.<br/>        :param proxies: (optional) The proxies dictionary to apply to the request.<br/>        :rtype: requests.Response<br/>        &quot;&quot;&quot;<br/>    <br/>        try:<br/>            conn = self.get_connection(request.url, proxies)<br/>        except LocationValueError as e:<br/>            raise InvalidURL(e, request=request)<br/>    <br/>        self.cert_verify(conn, request.url, verify, cert)<br/>        url = self.request_url(request, proxies)<br/>        self.add_headers(request, stream=stream, timeout=timeout, verify=verify, cert=cert, proxies=proxies)<br/>    <br/>        chunked = not (request.body is None or &#x27;Content-Length&#x27; in request.headers)<br/>    <br/>        if isinstance(timeout, tuple):<br/>            try:<br/>                connect, read = timeout<br/>                timeout = TimeoutSauce(connect=connect, read=read)<br/>            except ValueError as e:<br/>                # this may raise a string formatting error.<br/>                err = (&quot;Invalid timeout {}. Pass a (connect, read) &quot;<br/>                       &quot;timeout tuple, or a single float to set &quot;<br/>                       &quot;both timeouts to the same value&quot;.format(timeout))<br/>                raise ValueError(err)<br/>        elif isinstance(timeout, TimeoutSauce):<br/>            pass<br/>        else:<br/>            timeout = TimeoutSauce(connect=timeout, read=timeout)<br/>    <br/>        try:<br/>            if not chunked:<br/>                resp = conn.urlopen(<br/>                    method=request.method,<br/>                    url=url,<br/>                    body=request.body,<br/>                    headers=request.headers,<br/>                    redirect=False,<br/>                    assert_same_host=False,<br/>                    preload_content=False,<br/>                    decode_content=False,<br/>                    retries=self.max_retries,<br/>                    timeout=timeout<br/>                )<br/>    <br/>            # Send the request.<br/>            else:<br/>                if hasattr(conn, &#x27;proxy_pool&#x27;):<br/>                    conn = conn.proxy_pool<br/>    <br/>                low_conn = conn._get_conn(timeout=DEFAULT_POOL_TIMEOUT)<br/>    <br/>                try:<br/>                    low_conn.putrequest(request.method,<br/>                                        url,<br/>                                        skip_accept_encoding=True)<br/>    <br/>                    for header, value in request.headers.items():<br/>                        low_conn.putheader(header, value)<br/>    <br/>                    low_conn.endheaders()<br/>    <br/>                    for i in request.body:<br/>                        low_conn.send(hex(len(i))[2:].encode(&#x27;utf-8&#x27;))<br/>                        low_conn.send(b&#x27;\r\n&#x27;)<br/>                        low_conn.send(i)<br/>                        low_conn.send(b&#x27;\r\n&#x27;)<br/>                    low_conn.send(b&#x27;0\r\n\r\n&#x27;)<br/>    <br/>                    # Receive the response from the server<br/>                    try:<br/>                        # For Python 2.7, use buffering of HTTP responses<br/>                        r = low_conn.getresponse(buffering=True)<br/>                    except TypeError:<br/>                        # For compatibility with Python 3.3+<br/>                        r = low_conn.getresponse()<br/>    <br/>                    resp = HTTPResponse.from_httplib(<br/>                        r,<br/>                        pool=conn,<br/>                        connection=low_conn,<br/>                        preload_content=False,<br/>                        decode_content=False<br/>                    )<br/>                except:<br/>                    # If we hit any problems here, clean up the connection.<br/>                    # Then, reraise so that we can handle the actual exception.<br/>                    low_conn.close()<br/>                    raise<br/>    <br/>        except (ProtocolError, socket.error) as err:<br/>            raise ConnectionError(err, request=request)<br/>    <br/>        except MaxRetryError as e:<br/>            if isinstance(e.reason, ConnectTimeoutError):<br/>                # TODO: Remove this in 3.0.0: see #2811<br/>                if not isinstance(e.reason, NewConnectionError):<br/>                    raise ConnectTimeout(e, request=request)<br/>    <br/>            if isinstance(e.reason, ResponseError):<br/>                raise RetryError(e, request=request)<br/>    <br/>            if isinstance(e.reason, _ProxyError):<br/>                raise ProxyError(e, request=request)<br/>    <br/>            if isinstance(e.reason, _SSLError):<br/>                # This branch is for urllib3 v1.22 and later.<br/>                raise SSLError(e, request=request)<br/>    <br/>            raise ConnectionError(e, request=request)<br/>    <br/>        except ClosedPoolError as e:<br/>            raise ConnectionError(e, request=request)<br/>    <br/>        except _ProxyError as e:<br/>            raise ProxyError(e)<br/>    <br/>        except (_SSLError, _HTTPError) as e:<br/>            if isinstance(e, _SSLError):<br/>                # This branch is for urllib3 versions earlier than v1.22<br/>                raise SSLError(e, request=request)<br/>            elif isinstance(e, ReadTimeoutError):<br/>&gt;               raise ReadTimeout(e, request=request)<br/><span class="error">E               requests.exceptions.ReadTimeout: HTTPSConnectionPool(host=&#x27;jsonplaceholder.typicode.com&#x27;, port=443): Read timed out. (read timeout=10)</span><br/><br/>..\venv\lib\site-packages\requests\adapters.py:529: ReadTimeout[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:58:47.781 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_004 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_query_002
2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: get
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/comments
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: {&#x27;postId&#x27;: 1}
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:153 - 请求 data：None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:58:47.797 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:59:10.590 | ERROR    | fastpt.common.send_request:__requests_engin:72 - 请求异常: HTTPSConnectionPool(host=&#x27;jsonplaceholder.typicode.com&#x27;, port=443): Read timed out. (read timeout=10)
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:59:11.076 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="skipped results-table-row">
        <tr>
          <td class="col-result">Skipped</td>
          <td> 测试003 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_003[module: test, case_id: event_upload_004, case_desc: 测试文件上传, is_run: 0, method: post, env: None, url: https://api.anonfiles.com/upload, params: None, headers: None,data_type: json,data: None,files: {&quot;file&quot;:&quot;D:\\0000000000_test\\666.txt&quot;},sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">0.00</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>(&#x27;D:\\git_project\\automated_api_pytest\\fastpt\\utils\\request\\data_parse.py&#x27;, 40, &#x27;Skipped: 用例跳过&#x27;)[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:58:34.841 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_003 -----------------
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:58:34.841 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试001 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_001[1-2]</td>
          <td class="col-time">2022-07-24 08:59:26.245161</td>
          <td class="col-duration">0.08</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_001 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.testcase.test_project.test_api:test_001:31 - This is a demo&#x27;s test
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:57:53.396 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试003 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_003[module: test, case_id: event_query_001, case_desc: 测试直接请求, is_run: None, method: get, env: None, url: https://jsonplaceholder.typicode.com/posts/1, params: None, headers: None,data_type: None,data: None,files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">12.22</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:57:53.413 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_003 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_query_001
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: get
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/posts/1
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:153 - 请求 data：None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:57:53.413 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:58:05.499 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:57:53.413217
2022-07-24 16:58:05.499 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 200
2022-07-24 16:58:05.499 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 77.49 ms
2022-07-24 16:58:05.626 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:108 - test result: ----&gt; PASS
2022-07-24 16:58:05.626 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:113 - 写入excel测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:58:05.626 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试003 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_003[module: test, case_id: event_query_002, case_desc: 测试全局变量, is_run: None, method: get, env: ${test}, url: /comments, params: {&quot;postId&quot;:1}, headers: None,data_type: None,data: None,files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">12.72</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:58:05.642 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_003 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_query_002
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: get
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/comments
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: {&#x27;postId&#x27;: 1}
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:153 - 请求 data：None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:58:05.642 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:58:18.332 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:58:05.642026
2022-07-24 16:58:18.332 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 200
2022-07-24 16:58:18.332 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 679.103 ms
2022-07-24 16:58:18.364 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:108 - test result: ----&gt; PASS
2022-07-24 16:58:18.364 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:113 - 写入excel测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:58:18.364 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试003 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_003[module: test, case_id: event_create_003, case_desc: 测试post, is_run: None, method: post, env: ${test}, url: /posts, params: None, headers: None,data_type: json,data: {&quot;title&quot;:&quot;foo&quot;,&quot;body&quot;:&quot;${dev}&quot;,&quot;userId&quot;:1},files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">16.47</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:58:18.364 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_003 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_create_003
2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: post
2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/posts
2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: None
2022-07-24 16:58:18.364 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：json
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:log_request_up:155 - 请求 json: {&quot;title&quot;: &quot;foo&quot;, &quot;body&quot;: &quot;http://127.0.0.1:8080&quot;, &quot;userId&quot;: 1}
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:58:18.379 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:58:34.763 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:58:18.379809
2022-07-24 16:58:34.763 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 500
2022-07-24 16:58:34.763 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 385.726 ms
2022-07-24 16:58:34.810 | ERROR    | fastpt.common.excel_operate:write_excel_report:110 - test result: ----&gt; FAIL
2022-07-24 16:58:34.810 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:113 - 写入excel测试报告成功
2022-07-24 16:58:34.841 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:108 - test result: ----&gt; PASS
2022-07-24 16:58:34.841 | SUCCESS  | fastpt.common.excel_operate:write_excel_report:113 - 写入excel测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:58:34.841 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试004 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_004[module: test, case_id: event_query_001, case_desc: 测试直接请求, is_run: None, method: get, env: None, url: https://jsonplaceholder.typicode.com/posts/1, params: None, headers: None,data_type: None,data: None,files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">12.93</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:58:34.857 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_004 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_query_001
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: get
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/posts/1
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:153 - 请求 data：None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:58:34.857 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:58:34.857165
2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 200
2022-07-24 16:58:47.781 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 916.486 ms
2022-07-24 16:58:47.781 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:85 - test result: ----&gt; PASS
2022-07-24 16:58:47.781 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:90 - 写入yaml测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:58:47.781 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试004 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_004[module: test, case_id: event_create_003, case_desc: 测试hooks, is_run: None, method: post, env: None, url: https://jsonplaceholder.typicode.com/posts, params: None, headers: None,data_type: json,data: {&apos;title&apos;: &apos;foo&apos;, &apos;body&apos;: &apos;${random_phone()}&apos;, &apos;userId&apos;: 1},files: None,sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">13.07</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:59:11.091 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_004 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_create_003
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: post
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://jsonplaceholder.typicode.com/posts
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: None
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:59:11.091 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：json
2022-07-24 16:59:11.201 | INFO     | fastpt.common.send_request:log_request_up:155 - 请求 json: {&quot;title&quot;: &quot;foo&quot;, &quot;body&quot;: &quot;15984191964&quot;, &quot;userId&quot;: 1}
2022-07-24 16:59:11.201 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:59:11.201 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: None
2022-07-24 16:59:11.201 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:59:11.201 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:59:11.201345
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 500
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 946.868 ms
2022-07-24 16:59:24.160 | ERROR    | fastpt.common.yaml_operate:write_yaml_report:87 - test result: ----&gt; FAIL
2022-07-24 16:59:24.160 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:90 - 写入yaml测试报告成功
2022-07-24 16:59:24.160 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:85 - test result: ----&gt; PASS
2022-07-24 16:59:24.160 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:90 - 写入yaml测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:59:24.160 | INFO     | fastpt.conftest:log_end:63 - end 

<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td> 测试004 </td>
          <td class="col-name">testcase/test_project/test_api.py::TestDemo::test_004[module: test, case_id: event_upload_004, case_desc: 测试文件上传, is_run: None, method: post, env: None, url: https://api.anonfiles.com/upload, params: None, headers: None,data_type: json,data: None,files: {&apos;file&apos;: &apos;D:\\\\0000000000_test\\\\666.txt&apos;},sql: None,assert: None,]</td>
          <td class="col-time">2022-07-24 08:59:26.260786</td>
          <td class="col-duration">1.74</td></tr>
        <tr class="collapsed">
          <td class="extra" colspan="5">
            <div class="log">[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/>[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe[gw0] win32 -- Python 3.8.10 D:\git_project\automated_api_pytest\venv\Scripts\python.exe<br/> -----------------------------Captured stderr setup------------------------------ <br/>2022-07-24 16:59:24.160 | INFO     | fastpt.conftest:function_fixture:60 - ----------------- Running case: test_004 -----------------
<br/> ------------------------------Captured stderr call------------------------------ <br/>2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:145 - 用例 module: test
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:146 - 用例 case_id: event_upload_004
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:147 - 请求 method: post
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:148 - 请求 url: https://api.anonfiles.com/upload
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:149 - 请求 params: None
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:150 - 请求 headers: None
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:151 - 请求 data 类型：json
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:155 - 请求 json: None
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:156 - 请求 sql: None
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:157 - 请求 files: {&#x27;file&#x27;: &#x27;D:\\\\0000000000_test\\\\666.txt&#x27;}
2022-07-24 16:59:24.160 | INFO     | fastpt.common.send_request:log_request_up:158 - 请求 assert: None
2022-07-24 16:59:24.176 | INFO     | fastpt.common.send_request:__requests_engin:63 - 开始发送请求...
2022-07-24 16:59:25.885 | INFO     | fastpt.common.send_request:log_request_down:162 - 请求发送时间: 2022-07-24 16:59:24.176264
2022-07-24 16:59:25.885 | INFO     | fastpt.common.send_request:log_request_down:163 - 响应状态码: 200
2022-07-24 16:59:25.885 | INFO     | fastpt.common.send_request:log_request_down:164 - 响应时间: 721.39 ms
2022-07-24 16:59:25.901 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:85 - test result: ----&gt; PASS
2022-07-24 16:59:25.901 | SUCCESS  | fastpt.common.yaml_operate:write_yaml_report:90 - 写入yaml测试报告成功
<br/> ----------------------------Captured stderr teardown---------------------------- <br/>2022-07-24 16:59:25.901 | INFO     | fastpt.conftest:log_end:63 - end 

2022-07-24 16:59:25.901 | INFO     | fastpt.conftest:package_fixture:41 - 测试用例执行结束
<br/></div></td></tr></tbody></table></body></html>